src/main/daml/Daml/Finance/Interface/Instrument/Swap/Fpml/Types.daml

```diff
@@ -0,0 +1,214 @@
+-- Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module Daml.Finance.Interface.Instrument.Swap.Fpml.Types where
+
+import Daml.Finance.Interface.Instrument.Base.Instrument qualified as BaseInstrument (K)
+import Daml.Finance.Interface.Types.Common (InstrumentKey)
+import Daml.Finance.Interface.Types.Date.Calendar (BusinessDayConventionEnum)
+import Daml.Finance.Interface.Types.Date.DayCount (DayCountConventionEnum)
+import Daml.Finance.Interface.Types.Date.RollConvention
+
+data Fpml = Fpml
+  with
+    instrument : InstrumentKey
+      -- ^ The instrument's key.
+    description : Text
+      -- ^ The description of the swap.
+    swapStreams : [SwapStream]
+      -- ^ Each element describes a stream of swap payments, for example a regular fixed or floating rate.
+    issueDate : Date
+      -- ^ The issue date of the instrument.
+    issuerPartyRef : Text
+      -- ^ Used to the identify which counterparty is the issuer in the swapStream.
+    calendarDataProvider : Party
+      -- ^ The reference data provider to use for the holiday calendar.
+    currency : BaseInstrument.K
+      -- ^ The currency of the swap. For example, if the swap pays in USD this should be a USD cash instrument.
+    lastEventTimestamp : Time
+      -- ^ (Market) time of the last recorded lifecycle event. If no event has occurred yet, the time of creation should be used.
+  deriving (Eq, Show)
+
+data SwapStream = SwapStream
+  with
+    payerPartyReference : Text
+    calculationPeriodDates : CalculationPeriodDates
+    calculationPeriodAmount : CalculationPeriodAmount
+  deriving (Eq, Show)
+
+data CalculationPeriodDates = CalculationPeriodDates
+  with
+    effectiveDate : Date
+    terminationDate : Date
+    calculationPeriodDatesAdjustments : CalculationPeriodDatesAdjustments
+    firstRegularPeriodStartDate : Optional Date
+    lastRegularPeriodEndDate : Optional Date
+    calculationPeriodFrequency : CalculationPeriodFrequency
+  deriving (Eq, Show)
+
+data CalculationPeriodDatesAdjustments = CalculationPeriodDatesAdjustments
+  with
+    businessDayConvention : BusinessDayConventionEnum
+    businessCenters : [Text]
+  deriving (Eq, Show)
+
+-- | A type defining the frequency at which calculation
```

Comment:
Would you have a link to where the Fpml data model is defined, for my reference?

Reply:
This is the main link for the FpML `swap`: https://www.fpml.org/spec/fpml-5-11-3-lcwd-1/html/confirmation/schemaDocumentation/schemas/fpml-ird-5-11_xsd/complexTypes/Swap.html

The `swapStream` is clickable from that chart and there are annotations at the bottom of each page.